pkgs/development/python-modules/pytorch/default.nix

```diff
@@ -174,33 +201,44 @@ in buildPythonPackage rec {
     ninja
   ] ++ lib.optionals cudaSupport [ cudatoolkit_joined ];
 
-  buildInputs = [
-    numpy.blas
-  ] ++ lib.optionals cudaSupport [ cudnn magma nccl ]
+  buildInputs = [(if mklSupport then mkl else openblas)]
+    ++ lib.optionals cudaSupport [ cudnn magma nccl ]
     ++ lib.optionals stdenv.isLinux [ numactl ];
 
   propagatedBuildInputs = [
     cffi
     click
     numpy
     pyyaml
+    # the following are required for tensorboard support
+    pillow six future tensorflow-tensorboard protobuf
   ] ++ lib.optionals openMPISupport [ openmpi ]
-    ++ lib.optional (pythonOlder "3.5") typing
-    ++ lib.optionals tensorboardSupport [pillow six future tensorflow-tensorboard];
+    ++ lib.optional (pythonOlder "3.5") typing;
 
-  checkInputs = [ hypothesis ninja ];
+  checkInputs = [ hypothesis ninja psutil ];
 
   doCheck = false; # tests take a long time for channel release, so doCheck should be overridden only when developing
```

Comment:
How long are we talking?  Would be nice in general to run all the tests, particularly since this one is so complicated on variant ways to build and configure it.

Reply:
Back when I was running build matricies in pytorch-world, it could take anywhere from 2-8 hours depending on the configuration and machine.